sdk-lib/src/main/java/cash/z/ecc/android/sdk/jni/RustBackend.kt

```diff
@@ -154,9 +154,16 @@ internal class RustBackend private constructor(
         )
     }
 
+    override suspend fun writeBlockMetadata(blockMetadata: Array<String>) = withContext(SdkDispatchers.DATABASE_IO) {
```

Comment:
Indeed there will be a stronger type; `Array<String>` is a placeholder, because passing arrays of typed data across JNIN is more complex than I had time to implement today.